Sources/FoundationEssentials/Subprocess/Subprocess.swift

```diff
@@ -0,0 +1,163 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+import SystemPackage
+
+public struct Subprocess: Sendable {
```

Comment:
From this perspective, adding `Sendable` in the future seems possible, while taking it away would not be an option at all? In the case of uncertainty like this maybe it's better to be conservative and not add it at least until we're certain that the compiler features are sufficient to make this work well in the future?

Reply:
I don't believe `~Copyable` and `~Escapable` are mutually exclusive with `Sendable`, they represent fundamentally different concepts so even if `Subprocess` were designed to be `~Copyable`/`~Escapable` then it still can be `Sendable` in my understanding. Since mutable `~Copyable` types cannot have multiple references, I think I'd find it rare that a `~Copyable` type weren't `Sendable`

Reply:
You are right that those three things are not exclusive and we should think here about what we expect people to do. Do we expect them to share a Subprocess over multiple isolation regions? With the new region isolation proposals a lot has changed here.

I personally am not yet convinced that this should be Sendable since we have the standard output and error types which must not be called from multiple concurrent contexts in parallel.